lib/AST/NameLookup.cpp

```diff
@@ -980,6 +980,42 @@ SelfBounds swift::getSelfBoundsFromWhereClause(
                            SelfBoundsFromWhereClauseRequest{decl}, {});
 }
 
+SelfBounds SelfBoundsFromGenericSignatureRequest::evaluate(
+    Evaluator &evaluator, const ExtensionDecl *extDecl) const {
+  SelfBounds result;
+  if (!extDecl->hasComputedGenericSignature())
```

Comment:
This is brittle because it assumes someone else forced the computation by calling getGenericSignature() first. We want to trigger request computation lazily and not write code that depends on other requests having been executed first. You can just remove this conditional.